lib/client/node_e2e_test.go

```diff
@@ -141,6 +143,15 @@ var _ = testutils.E2eDatastoreDescribe("Node tests", testutils.DatastoreEtcdV2,
 			_, err = c.Nodes().Get(meta1)
 			Expect(err.Error()).To(Equal(errors.New("resource does not exist: Node(name=node1)").Error()))
 
+			// Node1 directory in etcd should be removed.
+			out, _ := exec.Command("curl", "http://127.0.0.1:2379/v2/keys/calico/v1/host/node1").Output()
+			Expect(string(out)).To(ContainSubstring("Key not found"))
+
+			// Node1 should not exist in etcd anymore.
+			dump, _ := exec.Command("curl", "http://127.0.0.1:2379/v2/keys/calico/?recursive=true").Output()
+			fmt.Println(string(dump))
+			Expect(string(dump)).NotTo(ContainSubstring("node1"))
+
 			// Delete node2 should not error.
```

Comment:
I think I'd rather see a separate test to explicitly test this stuff.  I think it better to implement a test that does the following:
-  Creates a node
-  Allocates some IP addresses for that node
-  Creates a WorkloadEndpoint on that node
-  Creates a BGPPeer on that node 
-  Deletes the node
-  Verifies all node configuration is removed from etcd.